ldap/src/integration-test/java/org/springframework/security/ldap/server/UnboundIdContainerTests.java

```diff
@@ -77,4 +82,50 @@ public void afterPropertiesSetWhenPortIsZeroThenRandomPortIsSelected() throws Ex
 		}
 	}
 
+	@Test
+	public void missingContextInInit() throws Exception {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, validLdifClassPath);
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+			assertThat(server.getPort()).isNotEqualTo(0);
+			failBecauseExceptionWasNotThrown(NullPointerException.class);
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(NullPointerException.class);
+		} finally {
+			server.destroy();
+		}
+	}
+
+	@Test
+	public void testMissingLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, "classpath:missing-file.ldif");
+		server.setApplicationContext(new GenericApplicationContext());
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(IllegalStateException.class);
+			assertThat(e.getMessage()).contains("Unable to load LDIF missing-file.ldif");
```

Comment:
This test is not functioning as it should.
We only reach the `catch` block if an exception is thrown.
If an exception is not thrown, then the test will still pass.
Please add `failBecauseExceptionWasNotThrown(IllegalStateException.class);` to the `try` block, as you have done for the other tests.